docs/develop/core.rst

```diff
@@ -0,0 +1,878 @@
+.. _coreruntime:
+
+The MicroPython Core Runtime
```

Comment:
I thinks the subtopics here could be split out into separate pages (e.g. compiler, qstrs, memory management, object model, etc.). This page is already quite long and the sections after qstrs don't seem to be very developed yet.

Reply:
okay, I will edit the outline accordingly.